src/backend/chat/custom/langchain.py

```diff
@@ -41,12 +44,16 @@ def chat(self, chat_request: LangchainChatRequest, **kwargs: Any) -> Any:
         tools = []
         for req_tool in chat_request.tools:
             tool = AVAILABLE_TOOLS.get(req_tool.name)
-            if tool:
+
+            if not tool:
+                logger.error(event=f"Failed to find tool {req_tool.name} during Langchain chat call.")
+
+            if not hasattr(tool.implementation, "to_langchain_tool"):
+                logger.error(event=f"Tool {req_tool.name} does not have to_langchain_tool method implemented.")
+
                 tools.append(tool.implementation().to_langchain_tool())
```

Comment:
Is the logic here correct? As it is now, it looks like the tool will only be appended if it doesn't have `to_langchain_tool` implemented. Do you want `continue` statements after each of the `logger.error` lines?

This also seems like a good area for a unit test.

Reply:
You're right, fixed it by adding continues and moving the append code outside of the ifs (the latter error seems to be a copy paste issue)